### ISSUE TITLE
Shade of northern-most row is not defined by the generated function

### ISSUE THREAD

**Issue by csemrau** [gd-codes/mc-pixelart-maker#29]
This issue is especially important for 3D maps: The shade of the first (northern-most) row is determined by the row directly north of it.

That row is not defined by the generated function. For a flat map, it simply must be the same height at the map itself and can be quickly added manually after creating the map art in a world.

For a 3D map, the desired shades determine which blocks in that row must be lower, higher, or same as the first map row. There's no way to manually construct that row based on information that is currently easily available.

**Comment by gd-codes** [comment]
This is a valid point - I assumed that in most cases the row would be higher than the one directly north of it, and displayed the preview accordingly in the fix for #15 - it avoids having to place an extra row of arbitrary blocks outside the map bounds.

Automating placing the extra row would be possible - if this seems like it impacts a number of maps; but if the differently shaded row isn't badly affecting the artwork (it's a small area on the edge), it seems neater to keep the map art structures within the volume that is actually mapped?

Example - sections of the top row stand out due to the height assumption:
<img width="100%" alt="image" src="https://github.com/gd-codes/mc-pixelart-maker/assets/45459312/21f7e607-90ba-4fd0-b8f5-e0d963bf0906">

**Comment by csemrau** [comment]
I agree with the assumption that a map is likely to be placed on a flat surface and any elevated block in the first row will have its lighter shade.

I have no opinion (yet) about whether the additional row should be part of the generated function or survival guide, since I'm still busy building my flat map in survival. For my image which is a noisy picture already, the one wrongly shaded row would not make a difference. 👍 
For the cat image from another issue, which has a plain background, it would likely be noticeable. But I am not building that, so... 🤷
